website/components/PropTable.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Table, TableHeader, Row, Cell } from '@leafygreen-ui/table';
+import { Subtitle, InlineCode } from '@leafygreen-ui/typography/dist';
+import PropDefinition from 'components/PropDefinition';
+import TypographyPropTable from 'components/TypographyPropTable';
+
+const subtitleBottomMargin = css`
+  margin-bottom: 24px;
+`;
+
+const tableBottomMargin = css`
+  margin-bottom: 56px;
+`;
+
+const tableHeaderNames: Array<'prop' | 'type' | 'description' | 'default'> = [
+  'prop',
+  'type',
+  'description',
+  'default',
+];
+
+/**
+ * Syntactic units in unist syntax trees are called nodes.
+ * Interface derived from Unified types
+ * See here: https://github.com/syntax-tree/unist#:~:text=Syntactic%20units%20in%20unist%20syntax,and%20implement%20the%20Node%20interface.
+ */
+export interface Node {
+  /**
+   * The variant of a node.
+   */
+  type: string;
+
+  /**
+   * Information from the ecosystem.
+   */
+  data?: {
+    [key: string]: unknown;
+  };
+
+  children?: Array<any>;
+
+  [key: string]: unknown;
+}
+
+interface TableDataInterface {
+  prop: string;
+  type: string;
+  description: string;
+  default: string;
+}
+
+interface RowElement {
+  type: string;
+  children?: Array<RowElement>;
+  value?: string;
+  position: { [key: string]: any };
+}
+
+function getTableData(rows: Array<RowElement>): Array<TableDataInterface> {
```

Comment:
semantically wouldn't one expect this function take an `Array<Row>`? in the code we only read `children` from each `row`. Looking at the code there's at least four distinct types we could have.

```tsx
interface Row {
  children: Array<CellRoot>;
}

interface CellRoot {
  children: Array<CellChild>;
}

type CellChild = { children: Array<CellLeaf>; } | CellLeaf;

type CellLeaf = { children?: undefined, value: string; };
```

Reply:
Implemented these types. Not sure I did it in the cleanest way possible 🤔 but let me know what you think